sample/kotlin/src/main/kotlin/com/datadog/android/sample/widget/WidgetIntentService.kt

```diff
@@ -0,0 +1,100 @@
+ */
+
+package com.datadog.android.sample.widget
+
+import android.app.IntentService
+import android.appwidget.AppWidgetManager
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.widget.RemoteViews
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumActionType
+import com.datadog.android.sample.R
+import com.datadog.android.sample.SampleApplication
+import okhttp3.Request
+import timber.log.Timber
+
+class WidgetIntentService : IntentService("WidgetIntentService") {
+
+    override fun onHandleIntent(intent: Intent?) {
+        when (intent?.action) {
+            LOAD_RANDOM_RESOURCE_ACTION -> {
+                handleActionWithRumContext(intent) {
+                    updateUIStatus(applicationContext, true)
+                    val okHttpClient = SampleApplication.getOkHttpClient(applicationContext)
+                    val builder = Request.Builder()
+                        .get()
+                        .url("https://www.datadoghq.com/")
+
+                    val request = builder.build()
+                    try {
+                        val response = okHttpClient.newCall(request).execute()
+                        val body = response.body()
+                        if (body != null) {
+                            val content: String = body.string()
+                            // Necessary to consume the response
+                            Timber.d("Response: $content")
+                        }
+                    } catch (e: Exception) {
+                        Timber.e("Error: ${e.message}")
+                    }
+                    updateUIStatus(applicationContext, false)
+                }
+            }
+            else -> {
+            }
+        }
+    }
+
+    private inline fun handleActionWithRumContext(
+        intent: Intent,
+        actionHandler: () -> Unit
+    ) {
+        val widgetName = intent.getStringExtra(WIDGET_NAME_ARG)
+        val widgetId = intent.getIntExtra(WIDGET_ID_ARG, 0)
+        val hasRumData = widgetId != 0 && widgetName != null
+        if (hasRumData) {
+            GlobalRum.get().startView(widgetId, widgetName ?: "DatadogWidget", emptyMap())
+            val clickedTargetName = intent.getStringExtra(WIDGET_CLICKED_TARGET_NAME)
+            if (clickedTargetName != null) {
+                GlobalRum.get().addUserAction(RumActionType.CLICK, clickedTargetName, emptyMap())
+            }
+        }
+        actionHandler()
+        if (hasRumData) {
+            GlobalRum.get().stopView(widgetId)
+        }
+    }
+
+    companion object {
+        const val LOAD_RANDOM_RESOURCE_ACTION = "load_random_resource"
+        const val WIDGET_ID_ARG = "widget_id"
+        const val WIDGET_NAME_ARG = "widget_name"
+        const val WIDGET_CLICKED_TARGET_NAME = "widget_clicked_target_name"
+
+        internal fun updateUIStatus(context: Context, isLoading: Boolean) {
```

Comment:
Again, why do you need to make it static ?

Reply:
wanted to follow their best practices...but I can put it inside, not really sure why they propose to use it like this.